bin/s2nc.c

```diff
@@ -86,7 +86,9 @@ static uint8_t unsafe_verify_host(const char *host_name, size_t host_name_len, v
 
     char *offset = strstr(host_name, "*.");
     if (offset) {
-        return (uint8_t)(strcasecmp(verify_data->trusted_host, offset + 2) == 0);
+        char *label = strchr(verify_data->trusted_host, '.');
+        return (uint8_t)(strcasecmp(verify_data->trusted_host, offset + 2) == 0) ||
```

Comment:
Is "example.com" allowed to match *.example.com?

Reply:
i don't think so - however the existing code would allow that.

i'd like to rewrite this function to be rfc compliant, however in the interest of time went with the simple fix.

we can create an issue to fix this,